### PR TITLE
Implement 999 per-actor production queue limit.

### PR DESCRIPTION
**Pull request by pchote**
This PR implements `ProductionQueue.QueueLimit` and `ProductionQueue.ItemLimit` to limit the number of items that can be queued in total and for each actor type respectively.

A per-item limit of 999 is defined to fix a game-breaking MP exploit while remaining large enough that nobody can reasonably complain.  This also fixes the "Training" production notification when trying to queue units that have already reached their build limit.

Testcase: Try to queue more than 999 of some actor type (or reduce the number, and try to build more than that).
Testcase: Set `QueueLimit: 1` and relive the frustration from the original game.
Testcase: Try to queue a second tanya while the first is already training (with instant build and all tech cheats disabled).  Notice that it now plays the correct audio.

### CODE REVIEW

**Comment by MustaphaTR** [comment]
D2k mod lacks `BuildingInProgress` notification and crashes when i reach 999. Original game didn't play any sound when you click a cameo when something else is being built.

Testcasses work fine on RA mod but i'm not sure that `enable tech` cheat should bypass these. It currently can: 
resim

**Comment by pchote** [comment]
> Testcasses work fine on RA mod but i'm not sure that enable tech cheat should bypass these. It currently can:

This was intentional, based on the precedent of it disabling build limits and production type / exit restrictions in addition to the prerequisites.  If there is enough objection then I have no real issue with changing it.

**Comment by pchote** [comment]
> Original game didn't play any sound when you click a cameo when something else is being built.

I guess we should expand #14892 to also cover sound notifications, and default all of these notifications to nothing.

**Comment by IceReaper** [comment]
As for the KKnD Mod i want to implement a behavior of stacking up to 10, and 11 will be "infinite". Does it make sense to implement my behavior into this pr too?

**Review by GraionDilach — COMMENTED** [review]
Looks good to me aside of the D2k crash, :+1: if there's a dummy sound until that other PR fixes the issue properly.

**Comment by pchote** [comment]
Fixed.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fc51fd4efb9acc19134455cb839eea1c7d11bfe6...94944f42234ece6086e68c8ae6112c95cc799177

File: OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs [hunk 1]
```diff
@@ -41,18 +41,29 @@ public class ProductionQueueInfo : ITraitInfo
...
 
 		[Desc("Notification played when production is complete.",
 			"The filename of the audio is defined per faction in notifications.yaml.")]
 		public readonly string ReadyAudio = "UnitReady";
 
-		[Desc("Notification played when you can't train another unit",
+		[Desc("Notification played when you can't train another actor",
 			"when the build limit exceeded or the exit is jammed.",
 			"The filename of the audio is defined per faction in notifications.yaml.")]
 		public readonly string BlockedAudio = "NoBuild";
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs]: Might this cause #15071? You removed `NoBuild` from the notifications but I don't see any overwrites for `BlockedAudio`.

**Pull request merged**